src/parser/types.rs

```diff
@@ -1161,7 +1162,33 @@ impl FunctionDef {
             ));
         }
 
-        let func = Self { sig, body, span };
+        let func = Self { sig, body, span, is_hint: false };
+
+        Ok(func)
+    }
+
+    /// Parse a hint function signature
+    pub fn parse_hint(ctx: &mut ParserCtx, tokens: &mut Tokens) -> Result<Self> {
```

Comment:
do you think it'd make sense to merge this with the parse function above?

Reply:
This new parser function is for hint function without a body.
So I think once we support the native hints, we can just remove this function, while the native hint function can get parsed using parser function above without changes.